src/api/manage/import.ts

```diff
@@ -1,32 +1,66 @@
-import type { PackedCourse, PackedEntity, PackedProgram, PackedTeacher } from "course-pack";
+import type {
+    CoursePack,
+    PackedCourse,
+    PackedEntity,
+    PackedProgram,
+    PackedTeacher
+} from "course-pack";
 import { verify as verify_course_pack } from "course-pack";
-import type { EndpointResponseBody, POST } from "unicourse";
+import type { EndpointRequestBody, EndpointResponseBody, POST } from "unicourse";
 import cuid from "cuid";
-import type { Entity } from "@unicourse-tw/prisma";
+import type { CourseProgram, Entity, Teacher } from "@unicourse-tw/prisma";
+import { down } from "@unicourse-tw/arborist";
 import UniRouter from "@/router";
 import { prisma } from "@/prisma";
 
 const router = new UniRouter();
 
 router.post("/", async ctx => {
-    const { body } = ctx.request;
+    const { body } = ctx.request as {
+        body: EndpointRequestBody<"manage/import", typeof POST>
+    };
     if (!body) {
         ctx.err("No body", { code: 400 });
         return;
     }
 
     const json = verify_course_pack(body);
 
+    let result: {
+        teachers: string[]
+        courses: string[]
+        programs: string[]
+    };
+    try {
+        if (body.scope) {
+            result = await scoped_resolver(json, body.scope);
+        } else {
+            result = await default_resolver(json);
+        }
+        ctx.ok<EndpointResponseBody<"manage/import", typeof POST>>(result);
+    } catch (err) {
+        if (err instanceof Error) {
+            ctx.err(err.message, { code: 400 });
+        }
```

Comment:
If `err` is not an `Error`. Maybe we should just throw it to avoid silent error?

Reply:
Um... that seems to be a good practice. It do help the application more robust.

> Note: I think all the errors must be implement as an `Error` or subclasses of `Error`, including "silent errors."
> We may need a new issue to fix this kind of errors.

Reply:
done in https://github.com/UniCourse-TW/Backend/pull/54/commits/54e9e51100b13a958ebe3609fa25a4b4a545baa4